nixos/modules/services/security/munge.nix

```diff
@@ -45,19 +45,25 @@ in
 
     systemd.services.munged = {
       wantedBy = [ "multi-user.target" ];
-      after = [ "network.target" ];
+      wants = [
+        "network-online.target"
+        "time-sync.target"
+      ];
+      after = [
+        "network-online.target"
+        "time-sync.target"
+      ];
 
       path = [ pkgs.munge pkgs.coreutils ];
 
       serviceConfig = {
         ExecStartPre = "+${pkgs.coreutils}/bin/chmod 0400 ${cfg.password}";
-        ExecStart = "${pkgs.munge}/bin/munged --syslog --key-file ${cfg.password}";
-        PIDFile = "/run/munge/munged.pid";
-        ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        ExecStart = "${pkgs.munge}/bin/munged --foreground --key-file ${cfg.password}";
         User = "munge";
         Group = "munge";
         StateDirectory = "munge";
         StateDirectoryMode = "0711";
+        Restart = "on-failure";
```

Comment:
Munge itself uses `Restart=on-abort` (https://github.com/dun/munge/blob/master/src/etc/munge.systemd.service.in)

Reply:
Yes, and I used the same in the previous version of the PR. I grepped for `Restart =` in nixpkgs to see what do people normally use. I had also skimmed through `man systemd.service` and I couldn't see reasons to use `on-abort` (restart on non-zero exit status) over `on-failure`:

> If set to on-failure, the service will be restarted when the process exits with a non-zero exit code, is terminated by a signal (including on core dump, but excluding the aforementioned four signals), when an operation (such as service reload) times out, and when the configured watchdog timeout is triggered

I wasn't sure if it's worth opening an issue upstream

Reply:
If in doubt, I'd prefer to use the upstream choice. It may be a matter of personal taste, but I don't like automatic restarts that hide software quality problems.

Reply:
In general I would also prefer to use the up-stream choice, unless there is a good reason not to. 
In this case, I have no strong opinion and we can leave it as-is right now.

Keep in mind though that any of these options would reload the service even if a misconfiguration would cause the abort/failure, causing a lot of noise in the logs. Thus the "Reload" option should only be used for system critical services, which munge is not IMHO.